diff --git a/Mf2/Parser.php b/Mf2/Parser.php
--- a/Mf2/Parser.php
+++ b/Mf2/Parser.php
@@ -404,8 +404,8 @@ class Parser {
 	}
 
 	private function elementPrefixParsed(\DOMElement $e, $prefix) {
-		if (!$this->parsed->contains($e))
-			$this->parsed->attach($e, array());
+		if (!$this->parsed->offsetExists($e))
+			$this->parsed->offsetSet($e, array());
 
 		$prefixes = $this->parsed[$e];
 		$prefixes[] = $prefix;
@@ -419,7 +419,7 @@ class Parser {
 	 * @return bool
 	 */
 	private function isElementParsed(\DOMElement $e, $prefix) {
-		if (!$this->parsed->contains($e)) {
+		if (!$this->parsed->offsetExists($e)) {
 			return false;
 		}
 
@@ -439,7 +439,7 @@ class Parser {
 	 * @return bool
 	 */
 	private function isElementUpgraded(\DOMElement $el, $property) {
-		if ( $this->upgraded->contains($el) ) {
+		if ( $this->upgraded->offsetExists($el) ) {
 			if ( in_array($property, $this->upgraded[$el]) ) {
 				return true;
 			}
@@ -937,7 +937,7 @@ class Parser {
 	 */
 	public function parseH(\DOMElement $e, $is_backcompat = false, $has_nested_mf = false) {
 		// If it’s already been parsed (e.g. is a child mf), skip
-		if ($this->parsed->contains($e)) {
+		if ($this->parsed->offsetExists($e)) {
 			return null;
 		}
 
@@ -1657,8 +1657,8 @@ class Parser {
 		}
 
 		// add element to list of upgraded elements
-		if ( !$this->upgraded->contains($el) ) {
-			$this->upgraded->attach($el, $property);
+		if ( !$this->upgraded->offsetExists($el) ) {
+			$this->upgraded->offsetSet($el, $property);
 		} else {
 			$this->upgraded[$el] = array_merge($this->upgraded[$el], $property);
 		}
